mdapi/MetricsDiscoveryHelper.cpp

```diff
@@ -0,0 +1,1142 @@
+                }
+                else if( setParams )
+                {
+                    DebugPrint("Skipped Group: %s MetricSet: %s MetricCount: %d API: %X, Category: %X\n",
+                        groupParams->SymbolName,
+                        setParams->SymbolName,
+                        setParams->MetricsCount,
+                        setParams->ApiMask,
+                        setParams->CategoryMask );
+                }
+            }
+        }
+    }
+
+    if (m_MetricSet == NULL)
+    {
+        DebugPrint("MetricSet is null\n");
+        return false;
+    }
+
+    m_Initialized = true;
+
+    DebugPrint("MetricsDiscoveryInit End\n");
+    return m_Initialized;
+}
+
+/************************************************************************/
+/* ActivateMetricSet                                                    */
+/************************************************************************/
+void MDHelper::ActivateMetricSet()
+{
+    if( !m_Initialized || !m_MetricSet ) return;
+
+    TCompletionCode res = m_MetricSet->Activate();
+    if( res != CC_OK ) DebugPrint("ActivateMetricSet failed!\n");
+}
+
+/************************************************************************/
+/* DeactivateMetricSet                                                  */
+/************************************************************************/
+void MDHelper::DeactivateMetricSet()
+{
+    if( !m_Initialized || !m_MetricSet ) return;
+
+    TCompletionCode res = m_MetricSet->Deactivate();
+    if( res != CC_OK ) DebugPrint("DeactivateMetricSet failed!\n");
+}
+
+/************************************************************************/
+/* SetMetricSetFiltering                                                */
+/************************************************************************/
+void MDHelper::SetMetricSetFiltering( TMetricApiType apiMask )
+{
+    if( !m_Initialized || !m_MetricSet ) return;
+
+    TCompletionCode res = m_MetricSet->SetApiFiltering( apiMask );
+    if( res != CC_OK ) DebugPrint("SetMetricSetFiltering failed!\n");
+}
+
+/************************************************************************/
+/* GetMetricsFromReport                                                 */
+/************************************************************************/
+void MDHelper::GetMetricsFromReport(
+    const char* pReportData,
+    std::vector<TTypedValue_1_0>& results,
+    std::vector<TTypedValue_1_0>& information )
+{
+    if( !m_Initialized || !m_MetricSet ) return;
+
+    const uint32_t metricsCount     = m_MetricSet->GetParams()->MetricsCount;
+    const uint32_t informationCount = m_MetricSet->GetParams()->InformationCount;
+
+    std::vector<TTypedValue_1_0>    deltaValues;
+
+    deltaValues.resize( metricsCount );
+    results.resize( metricsCount );
+    information.resize( informationCount );
+
+    // Read metrics from the report data and normalize:
+    ReadMetrics( pReportData, deltaValues );
```

Comment:
You can use IMetricSet_1_1::CalculateMetrics instead of ReadMetrics and NormalizeMetrics.

Reply:
You should :) 
(remember that calling SetApiFiltering is required prior to using CalculateMetrics())